csharp/test/Ice/ami/AllTests.cs

```diff
@@ -3842,6 +3842,36 @@ public static void allTests(global::Test.TestHelper helper, bool collocated)
                 }
                 output.WriteLine("ok");
 
+                output.Write("testing back pressure... ");
+                output.Flush();
+                {
+                    // Keep the 3 server thread pool threads busy.
+                    Task sleep1Task = p.sleepAsync(1000);
+                    Task sleep2Task = p.sleepAsync(1000);
+                    Task sleep3Task = p.sleepAsync(1000);
+                    bool canceled = false;
+                    using(var cts = new CancellationTokenSource(200))
+                    {
+                        try
+                        {
+                            var onewayProxy = (Test.TestIntfPrx)p.ice_oneway();
+
+                            // Sending should be canceled because the TCP receive buffer size on the server is set to
+                            // 50KB
+                            onewayProxy.opWithPayloadAsync(new byte[768 * 1024]).Wait(cts.Token);
```

Comment:
Why not use the cancel parameter of opWithPayloadAsync?

https://doc.zeroc.com/ice/3.7/language-mappings/c-sharp-mapping/client-side-slice-to-c-sharp-mapping/asynchronous-method-invocation-ami-in-c-sharp/ami-in-c-sharp-with-tasks#id-.AMIinCSharpwithTasksv3.7-CancelingAsynchronousRequestsinC#

Reply:
I tried but that's another weirdness of Ice for C#: the cancellation of an operation with the cancellation token doesn't follow the C# pattern. It works like the invocation timeout and like C++/Java it raises `InvocationCanceledException`. An operation which isn't being sent is not canceled until the payload is sent.